mm2src/mm2_p2p/src/behaviours/atomicdex.rs

```diff
@@ -321,14 +325,31 @@ impl AtomicDexBehaviour {
     fn spawn(&self, fut: impl Future<Output = ()> + Send + 'static) { self.runtime.spawn(fut) }
 
     fn process_cmd(&mut self, cmd: AdexBehaviourCmd) -> Result<(), AdexBehaviourError> {
+        let deprecated_netid_topic =
+            |netid| format!("{netid} netid is deprecated, the netid for main network is {DEFAULT_NETID}");
+
         match cmd {
             AdexBehaviourCmd::Subscribe { topic } => {
                 self.core.gossipsub.subscribe(&IdentTopic::new(topic))?;
             },
-            AdexBehaviourCmd::PublishMsg { topic, msg } => {
+            AdexBehaviourCmd::PublishMsg { mut topic, mut msg } => {
+                if DEPRECATED_NETID_LIST.contains(&self.netid) {
+                    topic = deprecated_netid_topic(self.netid);
+                    msg = Default::default();
+                };
+
                 self.core.gossipsub.publish(TopicHash::from_raw(topic), msg)?;
             },
-            AdexBehaviourCmd::PublishMsgFrom { topic, msg, from } => {
+            AdexBehaviourCmd::PublishMsgFrom {
+                mut topic,
+                mut msg,
+                from,
+            } => {
+                if DEPRECATED_NETID_LIST.contains(&self.netid) {
+                    topic = deprecated_netid_topic(self.netid);
+                    msg = Default::default();
+                };
```

Comment:
One question about your PR opening comment

> cut the communications on 7777 network with giving a informative message (e.g., "7777 netid is deprecated, the netid for main network is 8762")

I understand how this cuts communication, but how is this message shown?

Reply:
`topic = deprecated_netid_topic(self.netid);` this overrides the topic into the error message which becomes an invalid topic and gets printed out on both peers.

Reply:
ooh, I see now that in `publish_from` it will get printed in the debug logs. Do I understand this right?

Reply:
Mmm, not exactly. Just realized this wouldn't work either. Right now there seem to be no option for sending a notification message to the already connected peers. Let me think about that for a while to figure out something if possible.

Reply:
> Ok, notifying peers should be done as a patch which should be applied manually on the current main branch and then run as a seednode.

We can do it in a different branch that is based on main then, no need to merge it to main.

Reply:
I meant creating diff patch and using it manually, not merging it

Reply:
I will clarify both of these changes tomorrow in the migration channel.